cmake/add_tests.cmake

```diff
@@ -40,19 +40,24 @@ function (add_unit_test)
   target_compile_definitions(${ADD_UNIT_TEST_TARGET} PRIVATE TOP_SRCDIR="${PROJECT_SOURCE_DIR}")
   target_link_libraries(${ADD_UNIT_TEST_TARGET} ${ADD_UNIT_TEST_DEPENDS} syslog-ng)
   target_include_directories(${ADD_UNIT_TEST_TARGET} PUBLIC ${ADD_UNIT_TEST_INCLUDES})
-  if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-    set_property(TARGET ${ADD_UNIT_TEST_TARGET} APPEND PROPERTY LINK_FLAGS "-Wl,--no-as-needed")
+  if (NOT APPLE)
+    set_property(TARGET ${ADD_UNIT_TEST_TARGET} APPEND_STRING PROPERTY LINK_FLAGS " -Wl,--no-as-needed")
   endif()
 
   if (${ADD_UNIT_TEST_CRITERION})
     target_link_libraries(${ADD_UNIT_TEST_TARGET} ${CRITERION_LIBRARIES})
     target_include_directories(${ADD_UNIT_TEST_TARGET} PUBLIC ${CRITERION_INCLUDE_DIRS})
     set_property(TARGET ${ADD_UNIT_TEST_TARGET} PROPERTY POSITION_INDEPENDENT_CODE FALSE)
 
-    if (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-      # https://gitlab.kitware.com/cmake/cmake/issues/16561
-      set_property(TARGET ${ADD_UNIT_TEST_TARGET} APPEND PROPERTY LINK_FLAGS "-Wl,-no_pie")
+    # https://gitlab.kitware.com/cmake/cmake/issues/16561
+    include(CheckCCompilerFlag)
+    check_c_compiler_flag(-no-pie NO_PIE_AVAILABLE)
+    if (NO_PIE_AVAILABLE)
+      set_property(TARGET ${ADD_UNIT_TEST_TARGET} APPEND_STRING PROPERTY LINK_FLAGS " -no-pie")
```

Comment:
So in summary `set_property(TARGET ${ADD_UNIT_TEST_TARGET} PROPERTY POSITION_INDEPENDENT_CODE FALSE)` does not work with mac and gcc v>=6, hence the workaround. Does the POSITION_INDEPENDENT_CODE give anything after this workaround, or we could just delete that line?

Reply:
We could remove that line, it does not give anything to the tests.
However, the code below is just a workaround that we'll remove after a "few" years (:smiley:), so I would leave `POSITION_INDEPENDENT_CODE` here.

Reply:
Ok fine for me.